### PR TITLE
Implemented Clipboard API

### PR DESCRIPTION
**Pull request by jamesmcroft**
## PR summary

Changes have been made to introduce the Windows.ApplicationModel.DataTransfer.Clipboard like API for XPlat for Windows, Android and iOS.

The changes were tested using the project's sample applications. 

- [x] UWP
- [x] Android
- [x] iOS
- [ ] CI/CD

## PR checklist

- [ ] Tests have been added/updated and pass
- [x] Code styling has been run on all new source file changes
- [ ] Wiki documentation has been added
- [x] Code sample has been added
- [x] Contains **NO** breaking changes

No breaking changes made.

## References

#71

### CODE REVIEW

File: XPlat.ApplicationModel.DataTransfer/Clipboard.Android.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    using System.Threading.Tasks;
+    using Android.App;
+    using Android.Content;
+
+    /// <summary>Gets and sets information from the clipboard object.</summary>
+    public static class Clipboard
+    {
+        private const string PrimaryClipKey = "XPlat.ClipboardText";
+
+        public static event EventHandler<object> ContentChanged;
```
Comment (tom-made) [inline XPlat.ApplicationModel.DataTransfer/Clipboard.Android.cs]: This event is missing XML docs.

File: XPlat.ApplicationModel.DataTransfer/Clipboard.Windows.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#if WINDOWS_UWP
+namespace XPlat.ApplicationModel.DataTransfer
+{
+    using System;
+    using System.Threading.Tasks;
+
+    /// <summary>Gets and sets information from the clipboard object.</summary>
+    public static class Clipboard
+    {
+        public static event EventHandler<object> ContentChanged;
```
Comment (tom-made) [inline XPlat.ApplicationModel.DataTransfer/Clipboard.Windows.cs]: This event is missing XML docs.

File: XPlat.ApplicationModel.DataTransfer/Clipboard.iOS.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+{
+    using System;
+    using System.Threading.Tasks;
+
+    using UIKit;
+
+    /// <summary>Gets and sets information from the clipboard object.</summary>
+    public static class Clipboard
+    {
+        public static event EventHandler<object> ContentChanged;
```
Comment (tom-made) [inline XPlat.ApplicationModel.DataTransfer/Clipboard.iOS.cs]: This event is missing XML docs.

File: XPlat.ApplicationModel.DataTransfer/Clipboard.Android.cs (hunk 1 above)
Comment (jamesmcroft) [reply XPlat.ApplicationModel.DataTransfer/Clipboard.Android.cs]: Updated with the relevant summary tag.

File: XPlat.ApplicationModel.DataTransfer/Clipboard.Windows.cs (hunk 2 above)
Comment (jamesmcroft) [reply XPlat.ApplicationModel.DataTransfer/Clipboard.Windows.cs]: Also updated with the relevant summary tag here

File: XPlat.ApplicationModel.DataTransfer/Clipboard.iOS.cs (hunk 3 above)
Comment (jamesmcroft) [reply XPlat.ApplicationModel.DataTransfer/Clipboard.iOS.cs]: Again, updated the summary tag here

**Comment by jamesmcroft** [comment]
> Are we looking to extend the clipboard API to cover more than just string text?

@tom-made, on this one, the idea is that this has been setup in a similar way to the UWP API, so yes, we can extend it, but I'm looking to only implement the text copy here.

**Pull request merged**